subprojects/core/src/main/java/org/gradle/api/tasks/bundling/AbstractArchiveTask.java

```diff
@@ -120,7 +120,7 @@ public String getArchiveName() {
      */
     @Deprecated
     public void setArchiveName(String name) {
-        DeprecationLogger.deprecateProperty("archiveName").replaceWith("archiveFileName").nagUser();
+        DeprecationLogger.deprecateProperty("archiveName").replaceWith("archiveFileName").withDslReferenceForProperty(AbstractArchiveTask.class).nagUser();
```

Comment:
I think it was a mistake to not include the underlying type that has the deprecated property, so we should include that as part of the deprecateProperty call.